src/server/game/Accounts/AccountMgr.cpp

```diff
@@ -298,6 +298,40 @@ namespace AccountMgr
         return false;
     }
 
+    bool HasAccountFlag(uint32 accountId, uint32 flag)
+    {
+        LoginDatabasePreparedStatement* stmt = LoginDatabase.GetPreparedStatement(LOGIN_SEL_ACCOUNT_FLAG);
+        stmt->SetData(0, accountId);
+        if (PreparedQueryResult result = LoginDatabase.Query(stmt))
+        {
+            uint32 flags = (*result)[0].Get<uint32>();
+            return (flags & flag) != 0;
+        }
+
+        return false;
+    }
+
+    void UpdateAccountFlag(uint32 accountId, uint32 flag, bool remove /*= false*/)
+    {
+        LoginDatabasePreparedStatement* stmt = LoginDatabase.GetPreparedStatement(
+            remove ? LOGIN_UPD_REMOVE_ACCOUNT_FLAG : LOGIN_UPD_ADD_ACCOUNT_FLAG
+        );
+        stmt->SetData(0, flag);
+        stmt->SetData(1, accountId);
+        LoginDatabase.Execute(stmt);
+    }
+
+    void ValidateAccountFlags(uint32 accountId)
+    {
+        uint32 security = GetSecurity(accountId);
```

Comment:
GetSecurity() also queries the database for data we already have in the logon context